driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,401 @@
+        public void onUnregister(Cluster cluster) {
+
+        }
+
+        @Override
+        public void update() {
+            Map<Host, Integer> errorsSinceLastUpdate = new HashMap<Host, Integer>(errors.size());
+
+            // Update the error rates
+            for (Map.Entry<Host, Integer> errorsByHost : errors.entrySet()) {
+                int errorsNb = errorsByHost.getValue();
+                // TODO: we may miss errors between get and put...
+                errors.put(errorsByHost.getKey(), 0);
+                errorsSinceLastUpdate.put(errorsByHost.getKey(), errorsNb);
+
+                double errorsPerSecHostRate = ((double) (errorsNb * updatePeriod)) / 1000;
+
+                ErrorRecord err = errorsPerSec.get(errorsByHost.getKey());
+                if (err == null) {
+                    if (errorsPerSecHostRate >= errorsThresholdRate) {
+                        // Initialize a retry period so that the host will be considered invalid
+                        errorsPerSec.put(errorsByHost.getKey(), new ErrorRecord(errorsPerSecHostRate, retryPeriod));
+                    }
+                } else {
+                    err.setErrorsPerSecRate(errorsPerSecHostRate);
+                }
+            }
+
+            // Update the retry periods
+            for (Map.Entry<Host, ErrorRecord> errorsRecsByHost : errorsPerSec.entrySet()) {
+                Integer nbErrorsSinceLastUpdate = errorsSinceLastUpdate.get(errorsRecsByHost.getKey());
+                if (nbErrorsSinceLastUpdate != null && nbErrorsSinceLastUpdate > 0) {
+                    // Reset the retryPeriod if there has been new errors since the last update
+                    errorsRecsByHost.getValue().setRetryPeriod(retryPeriod);
+                    continue;
+                }
+
+                // for the rest only decrement the retryPeriod
+                ErrorRecord err = errorsRecsByHost.getValue();
+                if (err.getRetryPeriod() > 0) {
+                    err.setRetryPeriod(err.getRetryPeriod() - updatePeriod);
+                }
+
+            }
+        }
+
+        @Override
+        public boolean isHostValid(Host host) {
+            ErrorRecord errorsPerSecForHost = errorsPerSec.get(host);
+            if (errorsPerSecForHost == null) {
+                // assume there hasn't been errors since startup
+                // if there's no error, it's under threshold
+                return true;
+            } else {
+                /*
+                Host is valid if :
+                - its retryPeriod is <= 0, retryPeriod > 0 means the host have had errors recently.
+                - its error rate is strictly under the threshold.
+                 */
+                return errorsPerSecForHost.getRetryPeriod() <= 0 && errorsPerSecForHost.getErrorsPerSecRate() < errorsThresholdRate;
+            }
+        }
+
+        @Override
+        public boolean shouldAccountForError(Host host, Statement statement, Exception e) {
```

Comment:
Nice, so someone could extend a `PerHostErrorTracker` and simply override this method to filter out certain errors, neat!